PR второго этапа.

algmironov opened the pull request.

algmironov committed:
Написал несколько меню, кнопок, добавил Сервис для отчетов, его имплементацию и контроллер, пока с одним методом.
Так же перегрузил метод по созданию клавиатуры, чтобы он принимал на вход update вместо message.
И написал для класса с клавиатурами Javadoc.

OlgaLugacheva commented on src/main/java/pro/sky/telegrambot/repository/UserRepository.java:
```diff
@@ -5,4 +5,6 @@
 
 public interface UserRepository extends JpaRepository<User, Long> {
 
+    User findUserById(Long id);
```
зачем это вообще если есть метод в родителе finById ? 
давайте уберем, мне кажется, не нужен

OlgaLugacheva commented on src/main/java/pro/sky/telegrambot/repository/ReportsRepository.java:
```diff
@@ -3,6 +3,10 @@
+
 public interface ReportsRepository extends JpaRepository<Report, Long> {
 
+    Collection<Report> findAllByUserId(Long userId);
```
а точно? 
а точно не `Collection<Report> findAllByUser(Long userId);` ?

algmironov committed:
Поправил код из замечаний наставника,
Дописал и поправил код сохранения отчетов и фотографий.
Дописал и поправил код контроллера.
Написал документацию к контроллеру.
Добавил описание моделей и полей моделей для Swagger

Delegant commented on src/main/java/pro/sky/telegrambot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -36,24 +49,88 @@ public void init() {
         telegramBot.setUpdatesListener(this);
     }
 
+    private boolean hasCallbackQuery(Update update) {
```
мне кажется можно было проще написать

Delegant commented on src/main/java/pro/sky/telegrambot/listener/TelegramBotUpdatesListener.java:
```diff
@@ -36,24 +49,88 @@ public void init() {
+                logger.info("==== Processing update with callback: {}", update.callbackQuery().data());
+                telegramBot.execute(menuService.menuLoader(update, INFO_TEXT,  INFO_MENU));
+                break;
+            case (ABOUT_US_CASE):
```
в кейсах много повторяющегося кода. Думаю можно сократить.

Delegant commented on src/main/java/pro/sky/telegrambot/controller/ReportController.java:
```diff
@@ -0,0 +1,61 @@
+
+        Collection<ReportPicture> savedCollection = new ArrayList<ReportPicture>();
+        try {
+            savedCollection = reportService.savePictures(reportId, files);
```
возможно, лучше перенести обработку ошибки в сервис?

Delegant reviewed (approved):
по мне все хорошо, комментарии добавил, исключительно как свое мнение и для обратной связи, на истинность не претендую.
Так же, думаю файлы фоток карты лучше добавить в гитигнор.
